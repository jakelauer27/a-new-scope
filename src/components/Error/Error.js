import React from 'react'

export default function Error(props) {
  return (
    <div className='error-box'>
      <img className='error-darth' src='./images/darth.svg' alt='Darth Vader Error'></img>
      <h1 className='sith-happens'>Sith Happens...</h1>
      <h4 className='error-sub'>Oops, something seems to have gone wrong.</h4>
      <button className='error-sub error-button' onClick={() => props.changePage('landing')}>Click here to return to site.</button>
    </div>
  )
}