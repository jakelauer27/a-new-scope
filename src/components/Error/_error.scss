.error-box {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
}

.sith-happens {
  font-family: 'Jedi';
  color: #FCDD2B;
  font-size: 4rem;
  margin: 0 auto;
  text-align: center;
}

.error-sub {
  color: white;
  font-size: 2rem;
  letter-spacing: 1px;
  text-align: center;
  margin: 10px;
  font-family: 'Pathway Gothic One', sans-serif;
}

.error-darth {
  width: 400px;
  margin-top: 20px;
  animation: pulse 3s infinite ease;
}

@keyframes pulse {
  0%, 100% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.1)
  }
}
