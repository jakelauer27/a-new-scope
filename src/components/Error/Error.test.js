import React from 'react'
import { shallow } from 'enzyme'
import Error from './Error'

describe('Error', () => {
    const mockReturnToLanding = jest.fn()
    const mockSetError = jest.fn()
    const wrapper = shallow(<Error returnToLanding={mockReturnToLanding} setError={mockSetError}/>)
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should go back to the landing page on click', () => {
    wrapper.find('.error-button').simulate('click')
    expect(mockReturnToLanding).toBeCalled()
    expect(mockSetError).toBeCalled()
  })
})
