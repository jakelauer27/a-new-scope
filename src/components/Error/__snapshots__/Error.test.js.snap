// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    returnToLanding={[MockFunction]}
    setError={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Darth Vader Error"
          className="error-darth"
          src="./images/darth.svg"
        />,
        <h1
          className="sith-happens"
        >
          Sith Happens...
        </h1>,
        <h4
          className="error-sub"
        >
          Oops, something seems to have gone wrong.
        </h4>,
        <Link
          replace={false}
          to="/"
        >
          <button
            className="error-sub error-button"
            onClick={[Function]}
            type="submit"
          >
            Click here to return to site.
          </button>
        </Link>,
      ],
      "className": "error-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Darth Vader Error",
          "className": "error-darth",
          "src": "./images/darth.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sith Happens...",
          "className": "sith-happens",
        },
        "ref": null,
        "rendered": "Sith Happens...",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Oops, something seems to have gone wrong.",
          "className": "error-sub",
        },
        "ref": null,
        "rendered": "Oops, something seems to have gone wrong.",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="error-sub error-button"
            onClick={[Function]}
            type="submit"
          >
            Click here to return to site.
          </button>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click here to return to site.",
            "className": "error-sub error-button",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Click here to return to site.",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Darth Vader Error"
            className="error-darth"
            src="./images/darth.svg"
          />,
          <h1
            className="sith-happens"
          >
            Sith Happens...
          </h1>,
          <h4
            className="error-sub"
          >
            Oops, something seems to have gone wrong.
          </h4>,
          <Link
            replace={false}
            to="/"
          >
            <button
              className="error-sub error-button"
              onClick={[Function]}
              type="submit"
            >
              Click here to return to site.
            </button>
          </Link>,
        ],
        "className": "error-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Darth Vader Error",
            "className": "error-darth",
            "src": "./images/darth.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sith Happens...",
            "className": "sith-happens",
          },
          "ref": null,
          "rendered": "Sith Happens...",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Oops, something seems to have gone wrong.",
            "className": "error-sub",
          },
          "ref": null,
          "rendered": "Oops, something seems to have gone wrong.",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="error-sub error-button"
              onClick={[Function]}
              type="submit"
            >
              Click here to return to site.
            </button>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Click here to return to site.",
              "className": "error-sub error-button",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Click here to return to site.",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
